javers-core/src/main/java/org/javers/repository/api/QueryParamsBuilder.java

```diff
@@ -0,0 +1,82 @@
+package org.javers.repository.api;
+
+import org.javers.common.validation.Validate;
+import org.joda.time.LocalDateTime;
+
+/**
+ * @author michal wesolowski
+ */
+public class QueryParamsBuilder {
+    private int limit;
+    private LocalDateTime from;
+    private LocalDateTime to;
+
+    private QueryParamsBuilder(int limit) {
+        this.limit = limit;
+    };
+
+    /**
+     * Initializes builder with given limit
+     *
+     * @param limit choose reasonable limit (number of objects to fetch),
+     *              production database could contain more records than you expect
+     */
+    public static QueryParamsBuilder withLimit(int limit) {
+        checkLimit(limit);
+        return new QueryParamsBuilder(limit);
+    }
+
+    /**
+     * Initializes builder with param values from given queryParams instance
+     *
+     * @param queryParams instance to initialize builder param values from
+     */
+    public static QueryParamsBuilder initializeWith(QueryParams queryParams) {
+        Validate.argumentIsNotNull(queryParams);
+
+        QueryParamsBuilder builder = QueryParamsBuilder.withLimit(queryParams.getLimit());
+        if (queryParams.from().isPresent()) {
```

Comment:
Z technicznego punktu widzenia przejście na jdk8 było super. Pytanie co na to aktualni użytkownicy. Dopóki Javers musi wspierać jdk7 proponowałbym zostawić ten fragment jak jest.